File: Week 2/C++ Solutions/lcm.cpp
#include <iostream>

using std::cout;
using std::cin;

long gcd(long a, long b) {
  if (a > b) {a, b = b, a;}
  long r = a % b;

  while (r != 0) {
    a = b;
    b = r;
    r = a % b;
  }
  return b;
}

// NB: gcd(a, b) * lcm(a, b) = a * b

int main() {
  long a, b;
  cin >> a >> b;

  cout << (a * b) / gcd(a, b) << "\n";
}

File: Week 2/lcm.rb
n1, n2 = gets.chomp.split(/ /).map(&:to_i)

def gcd(a, b)
  a, b = b, a if a > b
  r = a % b
  while !r.zero?
    a = b
    b = r
    r = a % b
  end
  b
end

# FYI: lcm(a, b) * gcd(a, b) = a * b
puts (n1 * n2) / gcd(n1, n2)
